lib/helpers/message_helper.dart

```diff
@@ -318,80 +290,76 @@ class MessageHelper {
     } else if (![null, ""].contains(message.associatedMessageGuid)) {
-      // It's a reaction message, get the "sender"
+      // It's a reaction message, get the sender
       String? sender = message.isFromMe! ? "You" : ContactManager().getContactTitle(message.handle);
-      String translated = message.text ?? "";
-      if (!kIsWeb && !kIsDesktop) {
-        // translate the reaction text if necessary (to account for different
-        // locales on the tapback sender's phone
-        final languageIdentifier = GoogleMlKit.nlp.languageIdentifier();
-        final locale = Localizations.localeOf(Get.context!);
-        // check if the reaction text is in a different language
-        final splits = message.text!.split("“");
-        final reactionStr = sender + " " + splits.first;
-        final lang = await languageIdentifier.identifyLanguage(reactionStr);
-        // only translate if we have a valid language and it isn't the device's
-        // current language
-        if (lang != "und" && lang != locale.languageCode) {
-          final translateLanguageModelManager = GoogleMlKit.nlp.translateLanguageModelManager();
-          if (!await translateLanguageModelManager.isModelDownloaded(lang)) {
-            await translateLanguageModelManager.downloadModel(lang, isWifiRequired: false);
-          }
-          if (!await translateLanguageModelManager.isModelDownloaded(locale.languageCode)) {
-            await translateLanguageModelManager.downloadModel(locale.languageCode, isWifiRequired: false);
-          }
-          final onDeviceTranslator = GoogleMlKit.nlp.onDeviceTranslator(sourceLanguage: lang, targetLanguage: locale.languageCode);
-          translated = await onDeviceTranslator.translateText(reactionStr);
-          if (splits.length > 1) {
-            translated = translated + (translated.endsWith(" ") ? "“" : " “") + splits.sublist(1).join("“");
-          }
-          if (translated.startsWith(sender)) {
-            translated = translated.substring(sender.length + 1);
-          }
-          languageIdentifier.close();
-          onDeviceTranslator.close();
+      // fetch the associated message object
+      Message? associatedMessage = await Message.findOne({'guid': message.associatedMessageGuid});
+      if (associatedMessage != null) {
+        // grab the verb we'll use from the reactionToVerb map
+        String? verb = ReactionTypes.reactionToVerb[message.associatedMessageType];
+        // we need to check balloonBundleId first because for some reason
+        // game pigeon messages have the text "�"
+        if (associatedMessage.isInteractive()) {
+          return "$sender $verb ${getInteractiveText(associatedMessage)}";
+        // now we check if theres a subject or text and construct out message based off that
+        } else if (isNullOrEmpty(associatedMessage.text, trimString: true) == false
+            || isNullOrEmpty(associatedMessage.text, trimString: true) == false) {
```

Comment:
This condition can be shortened to `!isNullOrEmpty(associatedMessage.text, trimString: true) || !isNullOrEmpty(associatedMessage.text, trimString: true)`

Reply:
isNullOrEmpty returns an optional bool (not entirely sure why) so I'd have to unwrap the bool first, I'm not sure if its okay to assert isNullOrEmpty as not null

Reply:
Ah I forgot about that, yes you can assert it as not null. It should never return null, but for some reason one of the packages we use has its own `nullOrEmpty` functions return null for whatever reason.